src/state/opamEnv.ml

```diff
@@ -391,7 +391,7 @@ let init_file = function
   | SH_fish -> "init.fish"
 
 let complete_script = function
-  | SH_bash -> Some OpamScript.complete
+  | SH_sh -> Some OpamScript.complete
```

Comment:
Shouldn't this be `| SH_bash | SH_sh -> ... ?

I wonder if something is going wrong with shell detection, rather than that this function needs fixing?

Reply:
Yes, having looked further - I think it's more important to investigate why `OpamStd.Sys.guess_shell_compat` is returning `SH_sh` on your system rather than `SH_bash`. Are you able to debug that yourself, or would you like some pointers?

Reply:
Shell detection seems to work, I checked on MacOS 10.13.4 and Ubuntu 18.04, so I don't think it's that. From my understanding of the code, we're just generating scripts for every shell on init, not just the one in use. Note:  bash and sh are the same entry in shells_list:
https://github.com/ocaml/opam/blob/4f2753428d11ca021b3cb684559e8fa2f119c686/src/state/opamEnv.ml#L367-L369

The source of the problem is the following code:
https://github.com/ocaml/opam/blob/4f2753428d11ca021b3cb684559e8fa2f119c686/src/state/opamEnv.ml#L517-L523

When we loop through ```shells_list```, the ```filef``` and ```scriptf``` functions have to match on the same shell variant, but ```complete_file``` ignores ```SH_bash``` and ```complete_script``` ignores ```SH_sh```. The ```complete.sh``` script itself is set up to exit early if the shell is not bash:
https://github.com/ocaml/opam/blob/4f2753428d11ca021b3cb684559e8fa2f119c686/src/state/shellscripts/complete.sh#L1

Reply:
Oh, indeed, how silly of me! In which case, I wonder if just for completeness (excuse pun) sake it should be `| SH_bash | SH_sh ->` (cf. the others).